src/Microsoft.IdentityModel.Protocols.OpenIdConnect/OpenIdConnectProtocolValidator.cs

```diff
@@ -403,5 +551,40 @@ protected virtual void ValidateNonce(JwtSecurityToken jwt, OpenIdConnectProtocol
                 }
             }
         }
+
+        /// <summary>
+        /// Validates that the 'state' in message is valid.
+        /// </summary>
+        /// <param name="validationContext">a <see cref="OpenIdConnectProtocolValidationContext"/> that contains the 'state' to validate.</param>
+        /// <exception cref="ArgumentNullException">if 'validationContext' is null.</exception>
+        /// <exception cref="OpenIdConnectProtocolInvalidStateException">if 'state' in the context does not match the state in the message.</exception>
+        protected virtual void ValidateState(OpenIdConnectProtocolValidationContext validationContext)
+        {
+            if (!RequireState)
```

Comment:
We should validate the state when RequireState is false but ValidationContext.State and ValidationContext.Message.State exists. Similar to ValidateNonce. If state is sent is the request, it must be present in the response.